scripts/generate-worldoption.sh

```diff
@@ -0,0 +1,208 @@
+        echo ""
+    fi
+
+    # Construct JSON struct based on struct type
+    if [[ $struct == *"Enum"* ]]; then
+        if [[ $struct == *"DeathPenalty"* ]]; then
+            echo "{\"id\": null, \"value\": {\"value\": \"EPalOptionWorldDeathPenalty::$(type_cast "$struct" "$value")\", \"type\": \"EPalOptionWorldDeathPenalty\"}, \"type\": \"EnumProperty\"}"
+        elif [[ $struct == *"Difficulty"* ]]; then
+            echo "{\"id\": null, \"value\": {\"value\": \"EPalOptionWorldDifficulty::$(type_cast "$struct" "$value")\", \"type\": \"EPalOptionWorldDifficulty\"}, \"type\": \"EnumProperty\"}"
+        fi
+    else
+        echo "{\"id\": null, \"value\": $(type_cast "$struct" "$value"), \"type\": \"$struct_property\"}"
+    fi
+}
+
+# Function to generate JSON config
+generate_json_config() {
+    local json_config="{}"
+    local config=${1//,/$'\n'}
+
+    # Loop through each key-value pair
+    while IFS='=' read -r key value; do
+        config_properties="$(json_struct "$(config_to_struct_type "$key")" "$value")"
+
+        if [[ -n "$config_properties" ]]; then
+            json_config=$(echo "$json_config" | jq --arg key "$key" --argjson config_properties "$config_properties" '.[$key] = $config_properties')
+        fi
+    done <<< "$config"
+
+    echo "$json_config"
+}
+
+# Function to load Palworldsettings.ini
+load_palworldsettings() {
+    local path="$1"
+    local config
+    config=$(grep "OptionSettings" "$path")
+    if [[ -z "$config" ]]; then
+        echo "WorldOption Generator: Failed to get OptionSettings from PalWorldSettings.ini"
+        exit 1
+    fi
+
+    # Remove "OptionSettings=" and parentheses
+    config="${config#OptionSettings=}"
+    config="${config#\(}"
+    config="${config%\)}"
+
+    echo "$config"
+}
+
+# Python script for converting JSON to .sav
+convert_json_to_sav_python=$(
+cat <<EOF
+import sys
+import json
+from palworld_save_tools.gvas import GvasFile
+from palworld_save_tools.palsav import compress_gvas_to_sav
+from palworld_save_tools.paltypes import PALWORLD_CUSTOM_PROPERTIES
+
+gvas_file = GvasFile.load(json.loads(sys.stdin.read()))
+if ("Pal.PalWorldSaveGame" in gvas_file.header.save_game_class_name or "Pal.PalLocalWorldSaveGame" in gvas_file.header.save_game_class_name):
+    save_type = 0x32
+else:
+    save_type = 0x31
+sav_file = compress_gvas_to_sav(
+    gvas_file.write(PALWORLD_CUSTOM_PROPERTIES), save_type
+)
+sys.stdout.buffer.write(sav_file)
+EOF
+)
+
+# Function to convert JSON to .sav
+convert_json_to_sav() {
+    local json_data="$1"
+    local output_path="$2"
+
+    echo "WorldOption Generator: Compressing WorldOption to .sav"
+    if echo "$json_data" | python3 -c "$convert_json_to_sav_python" > "$output_path/WorldOption.sav"; then
+        echo "WorldOption Generator: Generated WorldOption.sav file to $output_path"
+        echo "Generating WorldOption.sav done!"
+    else
+        echo "WorldOption Generator: WorldOption.sav generation failed."
+    fi
+}
```

Comment:
Please don't include python code in this way.
Make a separate python file and call that.